Puc/v4p2/Plugin/UpdateChecker.php

```diff
@@ -399,6 +400,49 @@ public function addCheckForUpdatesLink($pluginMeta, $pluginFile) {
 		}
 
 		/**
+		 * Add a "View Details" link to the plugin row in the "Plugins" page. By default,
+		 * the new link will appear after the "Visit plugin site" link.
+		 *
+		 * You can change the link text by using the "puc_view_details_link-$slug" filter.
+		 * Returning an empty string from the filter will disable the link.
+		 *
+		 * @param array $pluginMeta Array of meta links.
+		 * @param string $pluginFile
+		 * @return array
+		 */
+		public function addViewDetailsLink($pluginMeta, $pluginFile) {
+			$isRelevant = ($pluginFile == $this->pluginFile)
+				|| (!empty($this->muPluginFile) && $pluginFile == $this->muPluginFile);
+
+			if ( $isRelevant && $this->userCanInstallUpdates() ) {
+				//Check if view-details is already among the links (because there is an update)
+				$viewDetailsExists = false;
+				foreach ($pluginMeta as $existingMeta) {
+					if (strpos($existingMeta, 'tab=plugin-information') !== false) {
```

Comment:
I'm not convinced that checking for "tab=plugin-information" is reliable enough. In theory, a plugin could have a link to a product page that or something similar that happens to include this query parameter.

How about this: Check if `$pluginData['slug']` is set. If WordPress knows the slug, it will automatically display the "View details" link. Of course, that could change in future WP versions, but this feature will need maintenance anyway to stay consistent with the core UI.

Also, please add spaces inside the outer parentheses, i.e.:
```php
( strpos($existingMeta, 'tab=plugin-information') !== false )
```